### PR TITLE
allow loading from path objects (py3.6 or above)

### PR DESCRIPTION
**Pull request by neutrinoceros**
this is a second try after #2392 
This PR is targeting yt-4.0 as the feature is much more easy to write for python 3.6 and above.

if we want to support python 3.5 too, an additional step is required to convert pathlike-non-str input to str.

### CODE REVIEW

**Comment by matthewturk** [comment]
@neutrinoceros are these test failures related to this PR?  I'm not quite sure that the ones about units are.

**Comment by neutrinoceros** [comment]
Hard to tell, I don't remember why I did not go through with this branch. I'll rebase and test it locally before pushing.

**Comment by neutrinoceros** [comment]
welp. Still failing. As I don't understand how to setup yt 4.0 right now (even in a separate env), I won't be able to solve those failures right now.

**Comment by neutrinoceros** [comment]
Since we're not supporting python 2 in the master branch anymore, I'll rebase and change target to master.

**Comment by neutrinoceros** [comment]
py2 build falling is expected. On the other hand, appveyor is also failing the tests, while they _do_ pass when I run locally. I don't understand what causes the failure there.

**Comment by Xarthisius** [comment]
Try adding this:

```diff
--- a/tests/tests.yaml
+++ b/tests/tests.yaml
@@ -140,5 +140,6 @@ answer_tests:
 other_tests:
   unittests:
      - '--exclude=test_mesh_slices'  # disable randomly failing test
+     - '--exclude=test_load_from_path'  # PY2
   cookbook:
      - 'doc/source/cookbook/tests/test_cookbook.py'
```

File: tests/tests.yaml [hunk 1]
```diff
@@ -140,5 +140,6 @@ answer_tests:
 other_tests:
   unittests:
      - '--exclude=test_mesh_slices'  # disable randomly failing test
+     - '--exclude=test_load_from_path'  # PY2
```
Comment (Xarthisius) [inline tests/tests.yaml:143]: oops, it was disabled for py3 too. It's apparently case sensitive, it should be "py2". Sorry for the wrong example.

Comment (neutrinoceros) [reply tests/tests.yaml]: Oh, nice catch !

**Comment by neutrinoceros** [comment]
Thanks to @Xarthisius , I think this is now ready for review !

**Pull request merged**